changelog-extractor.py

```diff
@@ -0,0 +1,144 @@
+"""Script parses changelog from git based on 2 latest tags, formates changelog to .md format, creates/updates release on gitlab, posts to slack
+"""
+import subprocess
+import os
+
+def getMdFormattedChangelog(newCommit, oldCommit) : 
+    #saving parameters this way to make sure that they are read properly on bash command
+    get_log_format = '%x1f'.join(['%b', '%h']) + '%x1e'
+    bashCommand = 'git --no-pager --git-dir=.git log --first-parent --format="%s"' % get_log_format   + " "+ newCommit+"..." + oldCommit
+    #getting raw changelog from git
+    rawChangelog = sendCommand(bashCommand).replace("\"", "")
+    return  ConvertChangelogTextToMd(rawChangelog, None)
+
+def buildCommandForTagNotes(reqType, cleanChangeLog, tag) : 
+    formattedProjectPath = os.environ['CI_PROJECT_PATH'].replace("/","%2F")  
+    return 'curl  -X '+reqType+'  --header "PRIVATE-TOKEN: $GITLAB_API_PRIVATE_TOKEN" -d description="'+cleanChangeLog+ '" https://git.smarpsocial.com/api/v4/projects/"'+ formattedProjectPath+ '"/repository/tags/"'+tag+'"/release'
+     
+def putTagNotesOnGitlab( cleanChangeLog  , tag):
+    # create a release notes just in case it hasnt'been created before
+    k = os.system(buildCommandForTagNotes("POST", cleanChangeLog, tag))
+    # updating just in case it had been created before
+    k = os.system(buildCommandForTagNotes("PUT", cleanChangeLog, tag))
+    print "\n"
+
+def sendCommand( bashCommand ):
```

Comment:
something like runCommand is less ambiguous than sendCommand

Reply:
done